tests/motion_setpoints.py

```diff
@@ -153,120 +164,138 @@ def test_GIVEN_2D_no_axis_WHEN_set_position_THEN_position_is_set_and_motor_moves
-        self.ca2DAxis.assert_that_pv_is("POSN", POSITION_SAMPLE2)
-        self.ca2DAxis.assert_that_pv_is("IPOSN", 1)
-        self.ca2DAxis.assert_that_pv_is("POSITIONED", 1)
+        self.motor_ca.set_pv_value("MTR0101", MOTOR_POSITION_SAMPLE2_COORD0 - 0.2)
+        self.motor_ca.assert_that_pv_is("MTR0101.RBV", MOTOR_POSITION_SAMPLE2_COORD0 - 0.2)
+        self.ca2D.assert_that_pv_is("POSN", POSITION_SAMPLE2)
+        self.ca2D.assert_that_pv_is("IPOSN", 1)
+        self.ca2D.assert_that_pv_is("POSITIONED", 1)
         """change tolerance, should not be in position"""
-        self.ca2DAxis.set_pv_value("TOLERENCE", 0.1)
-        self.ca2DAxis.assert_that_pv_is("POSN", "")
-        self.ca2DAxis.assert_that_pv_is("IPOSN", -1)
-        self.ca2DAxis.assert_that_pv_is("POSITIONED", 0)
+        self.ca2D.set_pv_value("TOLERENCE", 0.1)
+        self.ca2D.assert_that_pv_is("POSN", "")
+        self.ca2D.assert_that_pv_is("IPOSN", -1)
+        self.ca2D.assert_that_pv_is("POSITIONED", 0)
         """move motor back, should now be in position again"""
-        self.motor_ca.set_pv_value("MTR0", MOTOR_POSITION_SAMPLE2_COORD1)
-        self.motor_ca.assert_that_pv_is("MTR0.RBV", MOTOR_POSITION_SAMPLE2_COORD1)
-        self.ca2DAxis.assert_that_pv_is("POSN", POSITION_SAMPLE2)
-        self.ca2DAxis.assert_that_pv_is("IPOSN", 1)
-        self.ca2DAxis.assert_that_pv_is("POSITIONED", 1)
+        self.motor_ca.set_pv_value("MTR0101", MOTOR_POSITION_SAMPLE2_COORD0)
+        self.motor_ca.assert_that_pv_is("MTR0101.RBV", MOTOR_POSITION_SAMPLE2_COORD0)
+        self.ca2D.assert_that_pv_is("POSN", POSITION_SAMPLE2)
+        self.ca2D.assert_that_pv_is("IPOSN", 1)
+        self.ca2D.assert_that_pv_is("POSITIONED", 1)
+
+    def _test_alarm_propogates(self, channel_access, motor_num):
+        assert_alarm_state_of_posn(channel_access, motor_num, ChannelAccess.Alarms.NONE)
+
+        self.motor_ca.set_pv_value("MTR010{}.HLSV".format(motor_num), "MAJOR")
+        current_posn = self.motor_ca.get_pv_value("MTR010{}".format(motor_num))
+        self.motor_ca.set_pv_value("MTR010{}.HLM".format(motor_num), current_posn - 1)
+        self.motor_ca.assert_that_pv_alarm_is("MTR010{}".format(motor_num), ChannelAccess.Alarms.MAJOR)
+
+        assert_alarm_state_of_posn(channel_access, motor_num, ChannelAccess.Alarms.MAJOR)
+
+    def test_GIVEN_1D_WHEN_axis_in_alarm_THEN_position_in_alarm(self):
+        self._test_alarm_propogates(self.ca1D, 1)
+
+    def test_GIVEN_2D_WHEN_second_axis_in_alarm_THEN_position_in_alarm(self):
+        self._test_alarm_propogates(self.ca2D, 2)
+
+    def test_GIVEN_10D_WHEN_set_position_THEN_position_is_set_and_motor_moves_to_position(self):
+        def check_motor_positions(expected_coords, readback):
+            pv_suffix = ".RBV" if readback else ""
+            for index, coord in enumerate(expected_coords):
+                controller = 1 if index < 8 else 2
```

Comment:
Suggestion: Move 8 out to well named constant with comment pointing towards the use of 8 axes per galil ioc